VirusTotal.NET/VirusTotal.cs

```diff
@@ -164,12 +178,34 @@ public ScanResult ScanFile(Stream fileStream, string filename)
             if (string.IsNullOrWhiteSpace(filename))
                 throw new ArgumentException("You must provide a filename. Preferably the original filename.");
 
-            //https://www.virustotal.com/vtapi/v2/file/scan
-            RestRequest request = PrepareRequest("file/scan");
-            request.AddFile("file", fileStream.CopyTo, filename);
+            if (fileStream.Length > PUBLIC_FILE_SIZE_UPLOAD_LIMIT)
+            {
+                //assume we are getting a private upload url
+                //https://www.virustotal.com/vtapi/v2/file/scan/upload_url
+                RestRequest uploadRequest = PrepareRequest("file/scan/upload_url", Method.GET);
+                var bigFile = GetResults<LargeFileUpload>(uploadRequest);
+
+                var bigFileUri = new Uri(bigFile.UploadUrl);
+                RestRequest request = new RestRequest(bigFileUri, Method.POST);
+                request.AddFile("file", fileStream.CopyTo, filename);
+
+                var largeFileUploadClient = new RestClient(bigFileUri.Scheme + "://" + bigFileUri.Host);
+                largeFileUploadClient.FollowRedirects = true;
+                largeFileUploadClient.Proxy = _client.Proxy;
+                largeFileUploadClient.Timeout = _largeFileUploadTimeout;
+
+                //Output
+                return GetResults<ScanResult>(request, largeFileUploadClient);
```

Comment:
This will allow very large files to be uploaded, but VT will time out before it can process them. I have a restriction set at 500MB on my client. That seems to be the magic number.